au/code/au/apply_rational_magnitude_to_integral.hh

```diff
@@ -142,20 +148,27 @@ struct MaxNonOverflowingValueImplWhenNumFits<T, MagT, false> {
 //
 // Branch based on whether the numerator of `MagT` can fit in the promoted type of `T`.
 //
-template <typename T, typename MagT, MagRepresentationOutcome NumOutcome>
+template <typename T,
+          typename MagT,
+          MagRepresentationOutcome NumOutcome,
+          bool IsNegRatioAndUnsignedType>
 struct MaxNonOverflowingValueImpl;
 
+// For any situation where we're applying a negative factor to an unsigned type, simply short
+// circuit to set the max to zero.
+template <typename T, typename MagT, MagRepresentationOutcome NumOutcome>
+struct MaxNonOverflowingValueImpl<T, MagT, NumOutcome, true> : ValueIsZero<T> {};
+
 // If the numerator fits in the promoted type of `T`, delegate further based on whether the
 // denominator is bigger.
 template <typename T, typename MagT>
-struct MaxNonOverflowingValueImpl<T, MagT, MagRepresentationOutcome::OK>
-    : MaxNonOverflowingValueImplWhenNumFits<T, MagT, is_known_to_be_less_than_one(MagT{})> {};
+struct MaxNonOverflowingValueImpl<T, MagT, MagRepresentationOutcome::OK, false>
```

Comment:
I tend to avoid `bool` as a parameter type as it leads to this; at this call site what does `false` mean?  I usually get around this by making the parameter an `enum` even if that `enum` is just giving names to `true` and `false` that help the code be self-documenting at the call site.